packages/react-devtools-shared/src/devtools/views/Components/InspectedElementContext.js

```diff
@@ -145,14 +129,37 @@ function InspectedElementContextController({children}: Props) {
     [bridge, store],
   );
 
+  const inspectedElementCache = getCacheForType(createInspectedElementCache);
+
   const getInspectedElement = useCallback<GetInspectedElement>(
     (id: number) => {
       const element = store.getElementByID(id);
       if (element !== null) {
-        return resource.read(element);
-      } else {
-        return null;
+        const maybeInspectedElement = inspectedElementCache.get(element);
+        if (maybeInspectedElement !== undefined) {
+          return maybeInspectedElement;
+        } else {
+          const maybeThenable = inspectedElementThenables.get(element);
```

Comment:
I think this should also live in your cache. E.g. you can structure the cache as `Map { key => record }` where `record` is `{ status, value }` and `value` is either the result (when status is fulfilled) or error (for rejected) or thenable (for pending). See `react-fs` for example.

Reply:
Yeah, agreed. I need to do some cleanup here. This was just a first pass to see how the transition works. :)

(Note this is a DRAFT RP still.)